src/datatype/error.rs

```diff
@@ -27,12 +29,14 @@ pub enum Error {
     Io(IoError),
     JsonDecoder(JsonDecoderError),
     JsonEncoder(JsonEncoderError),
+    JsonParser(JsonParserError),
     Poison(String),
     Package(String),
     Parse(String),
     Recv(RecvError),
     SendEvent(SendError<Event>),
     SendInterpret(SendError<Interpret>),
+    SystemInfoError(String),
```

Comment:
Can we replace this with `SystemInfo` to keep clippy happy? (run `make clippy` for lint checks)